subs/pantry/src/Pantry/Types.hs

```diff
@@ -257,19 +285,54 @@ instance Display PackageLocation where
   display (PLImmutable loc) = display loc
   display (PLMutable fp) = fromString $ toFilePath $ resolvedAbsolute fp
 
+-- | Convert `PackageLocation` to its "raw" equivalent
+--
+-- @since 0.1.0.0
+toRawPL :: PackageLocation -> RawPackageLocation
+toRawPL (PLImmutable im) = RPLImmutable (toRawPLI im)
+toRawPL (PLMutable m) = RPLMutable m
```

Comment:
It's worth notice that the corresponding roundtrip loses info.
So if you start from a `RawPackageLocation`, then make it into a `PackageLocation` and then back, you'll lose information and that could make the error messages a bit annoying.
For error messages we should use `RawPackageLocation` if possible.

Reply:
Nah, that opens up an interesting situation: if for a package in `extra-deps` you don't set a concrete revision Stack is supposed to resolve it and work with that resolved exact location (maybe from a lock file which is to appear soon). If we won't keep a corresponding raw location then error message could be misleading: e.g. after Stack lock file was created package author could have created a new revision fixing dependency problem but Stack will complain about that previous resolved revision if there will be no measures against such a situation.

Reply:
Aha! Thanks